src/skin/qml/qmlcontrolproxy.h

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <QtQml>
+#include <memory>
+
+#include "control/controlproxy.h"
+
+namespace mixxx {
+namespace skin {
+namespace qml {
+
+class QmlControlProxy : public QObject {
+    Q_OBJECT
+// The REQUIRED flag only exists in Qt 5.14 and later.
+#if QT_VERSION >= QT_VERSION_CHECK(5, 14, 0)
+    Q_PROPERTY(QString group READ getGroup WRITE setGroup NOTIFY groupChanged REQUIRED)
+    Q_PROPERTY(QString key READ getKey WRITE setKey NOTIFY keyChanged REQUIRED)
+#else
+    Q_PROPERTY(QString group READ getGroup WRITE setGroup NOTIFY groupChanged)
+    Q_PROPERTY(QString key READ getKey WRITE setKey NOTIFY keyChanged)
+#endif
+    Q_PROPERTY(QString keyValid READ isKeyValid NOTIFY keyValidChanged)
+    Q_PROPERTY(QString initialized READ isInitialized NOTIFY initializedChanged)
+    Q_PROPERTY(double value READ getValue WRITE setValue NOTIFY valueChanged)
+    Q_PROPERTY(double parameter READ getParameter WRITE setParameter NOTIFY parameterChanged)
+
+  public:
+    QmlControlProxy(QObject* parent = nullptr);
+
+    void setGroup(const QString& group);
```

Comment:
Do we need to set group and key separately and thereby expose intermediate states?

Reply:
Yes. Let's say you declare a QML object like this:
```qml
import Mixxx 0.1 as MIxxx

Item {
    Mixxx.ControlProxy {
        id: mycontrol
        group: "[Channel1]"
        key: "play"
        
        onValueChanged: console.warn("value changed to" + mycontrol.value);
    }
}
```

What happens is that the object is instantiated, and then it calls `setGroup` and `setKey`. QML C++ objects can't take arguments in their constructors.